app/models/list.rb

```diff
@@ -2,27 +2,25 @@ class List < ApplicationRecord
   has_many :tasks, dependent: :destroy
   has_and_belongs_to_many :users
 
+  before_validation :generate_webhook_token
```

Comment:
I really liked the original `brefore_create` implementation, because you need not be concerned with the token, but couldn't figure out how to retain that feel while adding customisation and validation.
I did not know that `on: :create` was a thing for callback registration. That's exactly what I was looking for. I wanted pre-validation but only for creation.

Reply:
Actioned in 479b9b1. Memoisation still necessary otherwise it clobbers a custom token.

Reply:
Do you mean if you set a token ahead of creating the list? I'd probably prefer to see an early exit rather than memoisation. Memoisation makes me think it gets called multiple times, or it's lazy. It's technically neither in this case. What do you think?